keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1108 @@
+  of init and non-init containers will ensure enough resources are available to the containers.
+
+* **Option 2: Phased requests/limits - Pod-level cgroups are set after init containers are completed.**
+
+  This will require specifying requests/limits for init containers at container level if the peak resource requirement/usage of init containers is higher than 
+  app containers requirement/usage. Init containers will rely on their own limits. The pod-level cgroup will be set *AFTER* the execution of init containers are completed
+
+  Example pod spec:
+
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  spec:
+    resources:
+      requests:
+        memory: 100Mi
+        cpu: 2
+      limits:
+        memory: 200Mi
+        cpu: 3
+    initContainers:
+    - name: init-myservice
+      resources:
+        requests:
+          memory: 300Mi
+          cpu: 4
+        limits:
+          memory: 600Mi
+          cpu: 6
+    containers:
+    - name: container-1
+      resources:
+        requests:
+          memory: 50Mi
+          cpu: 1
+        limits:
+          memory: 100Mi
+          cpu: 2
+      - name: container-2
+        resources:
+          requests:
+            memory: 50Mi
+            cpu: 1
+          limits:
+            memory: 100Mi
+            cpu: 2 
+  ```
+
+  **Resources requests at pod-level**
+
+  * The scheduler will use the higher of init and pod-level requests values to find the suitable node for pod placement.
+    
+    From the example above, the scheduler will use:
+
+      max(init container resources requests, pod-level resources requests) 
+
+        = max (memory: 300Mi cpu: 4, memory: 100 Mi cpu: 2)
+
+        = (memory: 300 Mi, cpu: 4) to find a suitable node.
+
+  * The init containers will rely on its own requests specified in the PodSpec. 
+  * After the init containers execution is completed, kubelet will use the pod-level CPU request to set cpu.weight at pod-level cgroup.
+
+  **Resources limits at pod-level**
+
+  * The init containers will rely on their own limits specified in the PodSpec.
+  * After the init containers execution is completed, kubelet will use CPU and memory limits at pod-level, and translate them to cpu.max and memory.max cgroup settings
+  at pod level respectively. 
+
+  **Pros**
+
+  * Reduces the problem of overprovisioning.
+
+  **Cons**
+
+  * Complicates the definitions of QoS classes as this will require considering requests/limits for both initContainer and pod-level requests/limits to attach a 
+  QoS class with a pod. It might be worth setting dynamic QoS classes if this option were to be implemented. The QoS class in the initialization phase is based on
+  init containers requests/limits, while the QoS class after the initialization phase takes pod-level requests/limits into account. 
+  * Complicated to implement as it requires additional logic in kubelet to monitor init containers completion after which it will set/change pod level cgroup settings.
```

Comment:
Another con is that it's confusing as a dev to understand that initContainers follow a different logic and that I have to do the math above to understand the actual cpu/memory request.